### PR TITLE
Add ability/docs to run specs in parallel

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Add ability/docs to run specs in parallel, locally

#### Why
The are 10k+ specs that take approx 1 hour
to run locally in a single process.

This enables them to be run in parallel in
as many processes as the machine has cores.
On an 8 core machine it took ~15 minutes.

#### How
To install, prepare and run:
```
bundle
rake parallel:create
rake parallel:prepare
rake parallel:migrate # needed after each migration
rake parallel:spec
```
Note, feature specs not working and alot of setup and customization
needed it looks like

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -120,6 +122,9 @@
     end
   end
 
+  # mute noise for parallel tests
```
Comment (jsugarman) [inline spec/rails_helper.rb]: remove

**Review by kmahern — APPROVED** [review]
🥇 Looks good to me. I pulled the branch and tried it and ran the specs in around 16 minutes

**Comment by naseberry** [comment]
Looks good, In CircleCI it seems to take a minute less to run the suite.

**Comment by jsugarman** [comment]
> Looks good, In CircleCI it seems to take a minute less to run the suite.

That will probably be because it is optimises by spreading load more evenly across the processors by keep timings from previous runs. Can't do that with parallel_tests gem locally, at least not out of the box

**Pull request merged**